reaktive/src/jsMain/kotlin/com/badoo/reaktive/utils/queue/ArrayQueue.kt

```diff
@@ -0,0 +1,74 @@
+package com.badoo.reaktive.utils.queue
+
+internal class ArrayQueue<T> : Queue<T> {
```

Comment:
As perviously mentioned, would be nice to refactor in the future.
I find the head / tail confusing.

Reply:
It's a common implementation of array queue, almost copy-paste from Java ArrayDeque.